core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -313,6 +317,11 @@ public void commit() {
     }
   }
 
+  @Override
+  public long expiredSnapshotsCount() {
```

Comment:
Do we have existing examples where we expose state of the result via such a method?

Reply:
No existing examples. This PR is separated from https://github.com/apache/iceberg/pull/9584 which adds an option not to delete files in expire_snapshots. In that case, this count will be useful since all other results will be zero.